src/components/program-settings.js

```diff
@@ -139,94 +139,64 @@ class ProgramSettings extends React.Component {
             case 'eventsDownload':
-                e.target.value > maxValues.eventsDownload
-                    ? (e.target.value = maxValues.eventsDownload)
-                    : (e.target.value = valueInput)
+                value > maxValues.eventsDownload
+                    ? (value = parseInt(maxValues.eventsDownload))
+                    : (value = parseInt(value))
                 break
             case 'eventsDBTrimming':
-                e.target.value > maxValues.eventsDBTrimming
-                    ? (e.target.value = maxValues.eventsDBTrimming)
-                    : (e.target.value = valueInput)
+                value > maxValues.eventsDBTrimming
+                    ? (value = parseInt(maxValues.eventsDBTrimming))
+                    : (value = parseInt(value))
                 break
             default:
                 break
         }
+        return value
+    }
 
+    handleChange = e => {
+        e.preventDefault()
+        let { value } = e.target
+        value = this.chooseSetting(e.target.name, value)
         this.setState({
             ...this.state,
-            [e.target.name]: e.target.value,
+            [e.target.name]: value,
         })
         this.updateGlobal = true
     }
 
     handleChangeDialog = e => {
         e.preventDefault()
-        const valueInput = e.target.value
-        switch (e.target.name) {
-            case 'teiDownload':
-                e.target.value > maxValues.teiDownload
-                    ? (e.target.value = maxValues.teiDownload)
-                    : (e.target.value = valueInput)
-                break
-            case 'teiDBTrimming':
-                e.target.value > maxValues.teiDBTrimming
-                    ? (e.target.value = maxValues.teiDBTrimming)
-                    : (e.target.value = valueInput)
-                break
-            case 'teReservedDownload':
-                e.target.value > maxValues.teReservedDownload
-                    ? (e.target.value = maxValues.teReservedDownload)
-                    : (e.target.value = valueInput)
-                break
-            case 'teReservedDBTrimming':
-                e.target.value > maxValues.teReservedDBTrimming
-                    ? (e.target.value = maxValues.teReservedDBTrimming)
-                    : (e.target.value = valueInput)
-                break
-            case 'eventsDownload':
-                e.target.value > maxValues.eventsDownload
-                    ? (e.target.value = maxValues.eventsDownload)
-                    : (e.target.value = valueInput)
-                break
-            case 'eventsDBTrimming':
-                e.target.value > maxValues.eventsDBTrimming
-                    ? (e.target.value = maxValues.eventsDBTrimming)
-                    : (e.target.value = valueInput)
-                break
-            default:
-                break
-        }
-
+        let { value } = e.target
+        value = this.chooseSetting(e.target.name, value)
```

Comment:
Similarly, this doesn't need to be a mutable variable

Reply:
I'm removing the mutable variable and calling `this.chooseSetting` directly on line 194